pandas/core/reshape/merge.py

```diff
@@ -952,6 +975,51 @@ def _validate_specification(self):
         if len(self.right_on) != len(self.left_on):
             raise ValueError("len(right_on) must equal len(left_on)")
 
+    def _validate(self, validate):
+
+        # Check uniqueness of each
+        if self.left_index:
+            left_unique = not (self.orig_left.index.duplicated()).any()
+        else:
+            left_unique = MultiIndex.from_arrays(self.left_join_keys
+                                                 ).is_unique
+
+        if self.right_index:
+            right_unique = not (self.orig_right.index.duplicated()).any()
+        else:
+            right_unique = MultiIndex.from_arrays(self.right_join_keys
+                                                  ).is_unique
+
+        # Check valid arg
+        if validate not in ['one_to_one', '1:1',
+                            'one_to_many', '1:m',
+                            'many_to_one', 'm:1',
+                            'many_to_many', 'm:m']:
+
+            raise ValueError("Not a valid argument for validate")
+
+        # Check data integrity
+        if validate in ["one_to_one", "1:1"]:
+            if not left_unique or not right_unique:
+                raise ValueError("Merge keys are not unique in either left"
+                                 " or right dataset; not a one-to-one merge")
+            if not left_unique:
+                raise ValueError("Merge keys are not unique in left dataset;"
+                                 " not a one-to-one merge")
+            if not right_unique:
+                raise ValueError("Merge keys are not unique in right dataset;"
+                                 " not a one-to-one merge")
+
+        if validate in ["one_to_many", "1:m"]:
```

Comment:
if -> elif (the same below) (makes it clearer code-wise they are exclusive if statements)